vispy/visuals/colorbar.py

```diff
@@ -0,0 +1,398 @@
+
+        elif orientation == "left" or orientation == "right":
+            self._program = ModularProgram(VERT_SHADER, FRAG_SHADER_VERTICAL)
+        else:
+            raise ColorBarVisual._get_orientation_error(self._orientation)
+
+        tex_coords = np.array([[0, 0], [1, 0], [1, 1],
+                              [0, 0], [1, 1], [0, 1]],
+                              dtype=np.float32)
+
+        self._program.frag['color_transform'] = Function(self._cmap.glsl_map)
+        self._program['a_texcoord'] = tex_coords.astype(np.float32)
+
+        self._update()
+
+    def _update(self):
+        """Rebuilds the shaders, and repositions the objects
+           that are used internally by the ColorBarVisual
+        """
+
+        x, y = self._center_pos
+        halfw, halfh = self._halfdim
+
+        anchor_x, anchor_y = ColorBarVisual._get_anchors(self._orientation)
+        # create a new label if this is the first time this function
+        # is being called. Otherwise, just update the existing bar
+        if self._label is None:
+            self.label = TextVisual(text=self._label_str,
+                                    anchor_x=anchor_x,
+                                    anchor_y=anchor_y)
+        else:
+            self.label.text = self._label_str
+
+        # if this is the first time we're initializing ever
+        # i.e - from the constructor, then create the TextVisual objects
+        # otherwise, just change the _text_.
+        # this is SUPER IMPORTANT, since the user might have modified the
+        # ticks first to have say, a white color. And then he might have
+        # edited the ticks.
+        # This way, we retain any changes made by the user
+        # to the _ticks while still reflecting changes in _clim
+        if self._ticks == []:
+            self._ticks.append(TextVisual(str(self._clim[0]),
+                                          anchor_x=anchor_x,
+                                          anchor_y=anchor_y))
+            self._ticks.append(TextVisual(str(self._clim[1]),
+                                          anchor_x=anchor_x,
+                                          anchor_y=anchor_y))
+        else:
+            self._ticks[0].text = str(self._clim[0])
+            self._ticks[1].text = str(self._clim[1])
+
+        # Place the labels according to the given orientation
+        if self._orientation == "bottom":
+            text_x = x
+            text_y = y + _TEXT_PADDING_FACTOR * (halfh + self.border_width)
+
+            self._label.pos = text_x, text_y
+
+            # TODO, HACK: This should ideally be a single TextVisual
+            # However, one TextVisual with multiple strings
+            # does not seem to be working as of now. (See #981)
+            # https://github.com/vispy/vispy/issues/981
+            self._ticks[0].pos = x - halfw, text_y
+            self._ticks[1].pos = x + halfw, text_y
+
+        elif self._orientation == "top":
```

Comment:
usually there is some way to set up parameters (scale factors, variable names) in the branches, and then set all the attributes after the branching. see if that's possible here